source/server/listener_manager_impl.cc

```diff
@@ -211,27 +197,29 @@ ListenerImpl::ListenerImpl(const envoy::api::v2::Listener& config, ListenerManag
     ProtobufTypes::MessagePtr message =
         Config::Utility::translateToFactoryConfig(transport_socket, config_factory);
 
-    // Each transport socket factory owns one SslServerContext, we need to store them all in a
-    // vector since Ssl::ContextManager doesn't own SslServerContext. While transportSocketFacotry()
-    // always returns the first element of transport_socket_factories_, other transport socket
-    // factories are needed when the default Ssl::ServerContext updates SSL context based on
-    // ClientHello. This behavior is a workaround for initial SNI support before the full SNI based
-    // filter chain match is implemented.
-    transport_socket_factories_.emplace_back(config_factory.createTransportSocketFactory(
-        name_, sni_domains, skip_context_update, *message, *this));
-    ASSERT(transport_socket_factories_.back() != nullptr);
+    // TODO(PiotrSikora): reject multiple filter chains with the same matching criteria.
```

Comment:
Do you think we have adequate documentation on filter chain match precedence? E.g. if we have a CIDR match, do we consider the most specific filter chain the winner, or the first in the list of filter chains?

Reply:
CIDR match isn't implemented yet, but the idea is that the most specific match wins.

Reply:
Sure. I think if you can add some comments on what the end goal behavior is in the docs or API it would be great.